bolt/lib/Core/BinaryFunction.cpp

```diff
@@ -3744,6 +3744,16 @@ void BinaryFunction::postProcessBranches() {
   assert(validateCFG() && "invalid CFG");
 }
 
+bool BinaryFunction::canClone() const {
+  // For instrumentation, we need to restrict the execution to the rewritten
+  // version of the function.
+  if (opts::Instrument)
+    return false;
+
+  // Check for the presence of metadata that cannot be duplicated.
+  return !hasEHRanges() && !hasSDTMarker() && !hasPseudoProbe() && !hasORC();
```

Comment:
These checks seem quite restrictive (especially pseudo probes that are supposed to be in every function), what are the functions that can be cloned?

Reply:
I expect only tiny leaf function to be cloned, everything else should be patched.